user_management/src/user/views.py

```diff
@@ -3,63 +3,93 @@
+        validation_errors = []
 
-    @staticmethod
-    def is_valid_user(json_request):
         username = json_request.get('username')
-        if not SignUpView.is_valid_username(username):
-            return False
-        return True
+        email = json_request.get('email')
+        password = json_request.get('password')
+        valid_username, error_message_username = self.is_valid_username(username)
+        valid_email, error_message_email = self.is_valid_email(email)
+        valid_password, error_message_password = self.is_valid_password(password)
+
+        if not valid_username:
+            validation_errors.append(error_message_username)
+        if not valid_email:
+            validation_errors.append(error_message_email)
+        if not valid_password:
+            validation_errors.append(error_message_password)
+
+        return validation_errors
 
     @staticmethod
     def is_valid_username(username):
         if username is None:
-            return False
+            return False, "Username empty"
         if len(username) > 20:
-            return False
-        return True
+            return False, f"Username length {len(username)} > 20"
+        users = User.objects.filter(username=username)
+        if users.exists():
+            return False, "Username already taken"
+        return True, None
 
+    @staticmethod
+    def is_valid_email(email):
+        if email is None:
+            return False, "Email empty"
+        if len(email) > 50:
+            return False, f"Email length {len(email)} > 50"
+        if '@' not in email:
+            return False, "Email missing @"
+        if '.' not in email:
+            return False, "Email missing \".\" character"
+        if email.count('@') > 1:
+            return False, "Email contains more than one @ character"
+        return True, None
 
-class UserView(View):
     @staticmethod
-    def get(request, user_id):
-        if authenticate_request(request) is None:
-            return JsonResponse(data={}, status=401)
-        users = User.objects.filter(id=user_id)
-        if not users.exists():
-            return JsonResponse(data={}, status=404)
-        user = users.first()
-        data = {
-            'username': user.username,
-            'elo': user.elo,
-        }
-        return JsonResponse(data, status=200)
+    def is_valid_password(password):
+        if password is None:
+            return False, "Password empty"
+        if len(password) < 8:
+            return False, f"Password length {len(password)} < 8"
+        if not any(char.isupper() for char in password):
+            return False, "Password missing uppercase character"
+        if not any(char.isdigit() for char in password):
+            return False, "Password missing digit"
+        if not any(char in "!@#$%^&*()-_+=" for char in password):
+            return False, "Password missing special character"
+        return True, None
```

Comment:
Password max length not checked
Need to check if the password has lowercase letters as well